CellFabric/cell_fabric/grid.py

```diff
@@ -52,6 +52,24 @@ def inverseBounds( self, physical):
         else:
             return ((q,ge), (q,ge))
 
+    def snapToLegal(self, idx, direction):
```

Comment:
Not sure this works either. Can idx be greater than equal to self.n? If so we probably need to do some modular arithmetic.

Reply:
I am assuming idx is already a tuple with (major, minor) grid points. I do go to the next major grid if there isn't a legal minor gridpoint to be found.